rest/blip_client_test.go

```diff
@@ -514,30 +626,63 @@ func NewBlipTesterClientOptsWithRT(tb testing.TB, rt *RestTester, opts *BlipTest
 	if err != nil {
 		return nil, err
 	}
+
 	client.pullReplication.bt.avoidRestTesterClose = true
 	client.pushReplication.bt.avoidRestTesterClose = true
 
 	return client, nil
 }
 
+func (btc *BlipTesterClient) Close() {
+	for _, collectionReplicator := range btc.CollectionClients {
+		collectionReplicator.Close()
+	}
+}
+
+func (btc *BlipTesterClient) initCollectionReplication(collection string, collectionIdx int) error {
+	btcReplicator := &BlipTesterCollectionClient{
+		docs:                 make(map[string]map[string]*BodyMessagePair),
+		attachments:          make(map[string][]byte),
+		lastReplicatedRev:    make(map[string]string),
+		BlipTesterClientOpts: &btc.BlipTesterClientOpts,
+		pullReplication:      btc.pullReplication,
+		pushReplication:      btc.pushReplication,
+		rt:                   btc.rt,
+	}
+
+	btcReplicator.collection = collection
+	btcReplicator.collectionIdx = collectionIdx
+
+	btc.CollectionClients[collection] = btcReplicator
+	return nil
+}
+
+// DummyCollection() just returns the BlipTesterCollectionClient with empty key. This is used when not running in a
+// collection mode
+func (btc *BlipTesterClient) DummyCollection() *BlipTesterCollectionClient {
+	return btc.CollectionClients[""]
+}
+
+func (btc *BlipTesterClient) GetCollection(collection string) (*BlipTesterCollectionClient, error) {
```

Comment:
nitpick - but following gocb's naming conventions

```suggestion
func (btc *BlipTesterClient) Collection(collection string) (*BlipTesterCollectionClient, error) {
```